src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java

```diff
@@ -54,6 +71,36 @@ public DockerRegistryEndpoint(String url, String credentialsId) {
     }
 
     /**
+     * Parse the registry endpoint out of a registry:port/namespace/name:tag string. Credentials are set to the id
+     * passed. The url is built from registry:port into https://registry:port, the same way docker push does.
+     * 
+     * @param s
+     * @param credentialsId
+     *            can be null
```

Comment:
It was not obvious to me until I read the sources that this was not used as such, merely passed along to the `DockerRegistryEndpoint` constructor. Worth clarifying.